### PR TITLE
Removed Client ID

### PR DESCRIPTION
**Pull request by andrespelaezp**
### Summary of changes

- Remove Client ID from CoreConfig
- Use Access Token to fetch Client ID internally

### Checklist

~- [ ] Added a changelog entry~

@sshropshire 
@andrespelaezp

### CODE REVIEW

File: Sources/PayPalNativeCheckout/PayPalClient.swift [hunk 1]
```diff
@@ -11,52 +11,64 @@ public class PayPalClient {
...
     /// Present PayPal Paysheet and start a PayPal transaction
     /// - Parameters:
     ///   - request: the PayPalRequest for the transaction
     ///   - presentingViewController: the ViewController to present PayPalPaysheet on, if not provided, the Paysheet will be presented on your top-most ViewController
     ///   - completion: Completion block to handle buyer's approval, cancellation, and error.
-    public func start(request: PayPalRequest, presentingViewController: UIViewController? = nil) {
-        CheckoutFlow.set(config: config, returnURL: returnURL)
+    public func start(request: PayPalRequest, presentingViewController: UIViewController? = nil) async {
+        do {
+            let clientId = try await apiClient.getClientId()
```
Comment (jaxdesmarais) [inline Sources/PayPalNativeCheckout/PayPalClient.swift]:
```suggestion
            let clientID = try await apiClient.getClientID()
```

File: Sources/PayPalNativeCheckout/PayPalUIInterfaces/CheckoutProtocol.swift [hunk 2]
```diff
@@ -12,7 +12,7 @@ protocol CheckoutProtocol {
     typealias CancelCallback = () -> Void
     typealias ErrorCallback = (PayPalCheckoutErrorInfo) -> Void
 
-    static func set(config: CoreConfig, returnURL: String)
+    static func set(config: CoreConfig, clientId: String, returnURL: String)
```
Comment (jaxdesmarais) [inline Sources/PayPalNativeCheckout/PayPalUIInterfaces/CheckoutProtocol.swift]:
```suggestion
    static func set(config: CoreConfig, clientID: String, returnURL: String)
```

File: Sources/PayPalNativeCheckout/PayPalUIInterfaces/CheckoutProtocol.swift [hunk 3]
```diff
@@ -25,9 +25,9 @@ protocol CheckoutProtocol {
 
 extension Checkout: CheckoutProtocol {
 
-    static func set(config: CoreConfig, returnURL: String) {
+    static func set(config: CoreConfig, clientId: String, returnURL: String) {
```
Comment (jaxdesmarais) [inline Sources/PayPalNativeCheckout/PayPalUIInterfaces/CheckoutProtocol.swift]:
```suggestion
    static func set(config: CoreConfig, clientID: String, returnURL: String) {
```

File: Sources/PayPalNativeCheckout/PayPalUIInterfaces/CheckoutProtocol.swift [hunk 4]
```diff
@@ -25,9 +25,9 @@ protocol CheckoutProtocol {
...
         let nxoConfig = CheckoutConfig(
-            clientID: config.clientID,
+            clientID: clientId,
```
Comment (jaxdesmarais) [inline Sources/PayPalNativeCheckout/PayPalUIInterfaces/CheckoutProtocol.swift]:
```suggestion
            clientID: clientID,
```

File: Sources/PaymentsCore/GetClientIdRequest.swift [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import Foundation
+
+struct GetClientIdRequest: APIRequest {
+
+    typealias ResponseType = GetClientIdResponse
```
Comment (jaxdesmarais) [inline Sources/PaymentsCore/GetClientIdRequest.swift]:
```suggestion
    typealias ResponseType = GetClientIDResponse
```

File: Sources/PaymentsCore/GetClientIdRequest.swift [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+    let path: String = "v1/oauth2/token"
+    let token: String
+    var method: HTTPMethod = .get
+    var body: Data?
+
+    var headers: [HTTPHeader: String] {
+        return [
```
Comment (jaxdesmarais) [inline Sources/PaymentsCore/GetClientIdRequest.swift]: Can remove the return keyword and rely on implicit return

File: Sources/PaymentsCore/GetClientIdResponse.swift [hunk 7]
```diff
@@ -0,0 +1,4 @@
+struct GetClientIdResponse: Decodable {
+
+    let clientId: String
```
Comment (jaxdesmarais) [inline Sources/PaymentsCore/GetClientIdResponse.swift]:
```suggestion
    let clientID: String
```

File: Sources/PaymentsCore/Networking/APIClient.swift [hunk 8]
```diff
@@ -41,4 +41,10 @@ public final class APIClient {
             throw APIClientError.serverResponseError(errorData.readableDescription)
         }
     }
+
+    public func getClientId() async throws -> String {
```
Comment (jaxdesmarais) [inline Sources/PaymentsCore/Networking/APIClient.swift]:
```suggestion
    public func getClientID() async throws -> String {
```

**Review by jaxdesmarais — COMMENTED** [review]
Mostly nits around updating `Id` to `ID`

**Comment by wallportillo** [comment]
After @jaxdesmarais feedback do we need more feedback on this @andrespelaezp?

File: Sources/PaymentsCore/GetClientIdRequest.swift (hunk 6 above)
Comment (sshropshire) [reply Sources/PaymentsCore/GetClientIdRequest.swift]: Should be all good!

File: Sources/PayPalNativeCheckout/PayPalClient.swift [hunk 9]
```diff
@@ -11,34 +11,52 @@ public class PayPalClient {
...
     }
 
     /// Present PayPal Paysheet and start a PayPal transaction
     /// - Parameters:
     ///   - request: the PayPalRequest for the transaction
     ///   - presentingViewController: the ViewController to present PayPalPaysheet on, if not provided, the Paysheet will be presented on your top-most ViewController
     ///   - completion: Completion block to handle buyer's approval, cancellation, and error.
-    public func start(request: PayPalRequest, presentingViewController: UIViewController? = nil) {
-        CheckoutFlow.set(config: config)
+    public func start(request: PayPalRequest, presentingViewController pvc: UIViewController? = nil) async {
```
Comment (jaxdesmarais) [inline Sources/PayPalNativeCheckout/PayPalClient.swift]: `pvc` feels like a more confusing name for this internal function parameter. Can we just use `presentingViewController`?

File: Sources/PaymentsCore/GetClientIDResponse.swift [hunk 10]
```diff
@@ -0,0 +1,4 @@
+struct GetClientIDResponse: Decodable {
+
+    let clientId: String
```
Comment (jaxdesmarais) [inline Sources/PaymentsCore/GetClientIDResponse.swift]:
```suggestion
    let clientID: String
```

File: Sources/PaymentsCore/Networking/APIClient.swift [hunk 11]
```diff
@@ -41,4 +41,10 @@ public final class APIClient {
             throw APIClientError.serverResponseError(errorData.readableDescription)
         }
     }
+
+    public func getClientID() async throws -> String {
+        let request = GetClientIDRequest(token: coreConfig.accessToken)
+        let (response, _) = try await fetch(endpoint: request)
+        return response.clientId
```
Comment (jaxdesmarais) [inline Sources/PaymentsCore/Networking/APIClient.swift]:
```suggestion
        return response.clientID
```

**Review by jaxdesmarais — APPROVED** [review]
Couple of very tiny cleanup comments but otherwise this looks great!

File: Sources/PaymentsCore/GetClientIDResponse.swift (hunk 10 above)
Comment (sshropshire) [reply Sources/PaymentsCore/GetClientIDResponse.swift]: Yeah this one's interesting I can't find a way to get it to work with `Decodable`. The JSON property name is `client_id` and we use `.convertFromSnakeCase` in the decoder.

Comment (jaxdesmarais) [reply Sources/PaymentsCore/GetClientIDResponse.swift]: Ah, forgot that we used `APIClientDecoder` for all decoding. Should be able to add a custom coding key to solve that issue though within the `GetClientIDResponse` struct:
```
struct GetClientIDResponse: Decodable {

    private enum CodingKeys: String, CodingKey {
        case clientID = "clientId"
    }

    let clientID: String
}
```

Comment (sshropshire) [reply Sources/PaymentsCore/GetClientIDResponse.swift]: Ah nice that worked!

**Review by jcnoriega — APPROVED** [review]

**Review by kaylagalway — APPROVED** [review]
lg :+1:

**Review by mattwylder — APPROVED** [review]

**Pull request merged**